crates/ruma-common/src/events/poll/end.rs

```diff
@@ -1,42 +1,95 @@
 pub struct PollEndEventContent {
-    /// The poll end content of the message.
-    #[serde(rename = "org.matrix.msc3381.poll.end", alias = "m.poll.end")]
-    pub poll_end: PollEndContent,
+    /// The text representation of the results.
+    #[serde(rename = "org.matrix.msc1767.text")]
+    pub text: TextContentBlock,
+
+    /// The sender's perspective of the results.
+    #[serde(
+        rename = "org.matrix.msc3381.v2.poll.results",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub poll_results: Option<PollResultsContentBlock>,
+
+    /// Whether this message is automated.
+    #[cfg(feature = "unstable-msc3955")]
+    #[serde(
+        default,
+        skip_serializing_if = "crate::serde::is_default",
+        rename = "org.matrix.msc1767.automated"
+    )]
+    pub automated: bool,
 
     /// Information about the poll start event this responds to.
     #[serde(rename = "m.relates_to")]
     pub relates_to: Reference,
 }
 
 impl PollEndEventContent {
-    /// Creates a new `PollEndEventContent` that responds to the given poll start event ID,
-    /// with the given poll end content.
-    pub fn new(poll_end: PollEndContent, poll_start_id: OwnedEventId) -> Self {
-        Self { poll_end, relates_to: Reference::new(poll_start_id) }
+    /// Creates a new `PollEndEventContent` with the given fallback representation and
+    /// that responds to the given poll start event ID.
+    pub fn new(text: TextContentBlock, poll_start_id: OwnedEventId) -> Self {
+        Self {
+            text,
+            poll_results: None,
+            #[cfg(feature = "unstable-msc3955")]
+            automated: false,
+            relates_to: Reference::new(poll_start_id),
+        }
+    }
+
+    /// Creates a new `PollEndEventContent` with the given plain text fallback representation and
+    /// that responds to the given poll start event ID.
+    pub fn with_plain_text(plain_text: impl Into<String>, poll_start_id: OwnedEventId) -> Self {
+        Self {
+            text: TextContentBlock::plain(plain_text),
+            poll_results: None,
+            #[cfg(feature = "unstable-msc3955")]
+            automated: false,
+            relates_to: Reference::new(poll_start_id),
+        }
     }
 }
 
-/// Poll end content.
+/// A block for the results of a poll.
 ///
-/// This is currently empty.
+/// This is a map of answer ID to number of votes.
 #[derive(Clone, Debug, Default, Serialize, Deserialize)]
 #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
-pub struct PollEndContent {}
+pub struct PollResultsContentBlock(BTreeMap<String, UInt>);
+
+impl From<BTreeMap<String, UInt>> for PollResultsContentBlock {
+    fn from(value: BTreeMap<String, UInt>) -> Self {
+        Self(value)
+    }
+}
+
+impl FromIterator<(String, UInt)> for PollResultsContentBlock {
+    fn from_iter<T: IntoIterator<Item = (String, UInt)>>(iter: T) -> Self {
+        Self(iter.into_iter().collect())
```

Comment:
```suggestion
        Self(BTreeMap::from_iter(iter))
```

Reply:
If there is a `FromIterator` impl, a corresponding `IntoIterator` would be good too, right?